git-cliff-core/src/config.rs

```diff
@@ -469,6 +471,44 @@ impl Config {
             .build()?
             .try_deserialize()?)
     }
+
+    /// Find a special config path on macOS.
+    ///
+    /// The `dirs` crate ignores the `XDG_CONFIG_HOME` env var on macOS and only considers
+    /// `Library/Application Support` as the config dir, which is primarily used by GUI apps.
+    ///
+    /// This function determines the config path and honors the `XDG_CONFIG_HOME` env var.
+    /// If it is not set, it will fall back to `~/.config`
+    #[cfg(target_os = "macos")]
+    pub fn retrieve_xdg_config_on_macos() -> PathBuf {
+        let config_dir = std::env::var("XDG_CONFIG_HOME").map_or_else(
+            |_| dirs::home_dir().unwrap_or_default().join(".config"),
+            PathBuf::from,
+        );
+        config_dir.join("git-cliff")
+    }
+
+    /// Find the path of the config file.
+    ///
+    /// If the config file is not found in its standard locations, None is returned.
+    pub fn retrieve_config_path() -> Option<PathBuf> {
+        let mut path: Option<PathBuf> = None;
```

Comment:
Instead of assigning to a variable, can you return `Some(T)` whenever a path is found? This way we can also get rid of `break` as well.

I'm not sure if early-return is idiomatic or not, but I'd like to have it here.

Reply:
done